src/utils/common.ts

```diff
@@ -290,34 +293,65 @@ export async function estimateTotalGasRequiredByUnsignedTransaction(
 }
 
 /**
- * Create an unsigned transaction of a fillRelay contract call
- * @param spokePool The specific spokepool that will populate this tx
- * @param destinationTokenAddress A valid ERC20 token (system-wide default is UDSC)
- * @param simulatedRelayerAddress The relayer address that relays this transaction
- * @returns A populated (but unsigned) transaction that can be signed/sent or used for estimating gas costs
+ * Create an unsigned transaction to fill a relay. This function is used to simulate the gas cost of filling a relay.
+ * @param spokePool A valid SpokePool contract instance
+ * @param fillToSimulate The fill that this function will use to populate the unsigned transaction
+ * @returns An unsigned transaction that can be used to simulate the gas cost of filling a relay
  */
-export async function createUnsignedFillRelayTransaction(
+export async function createUnsignedFillRelayTransactionFromDeposit(
   spokePool: SpokePool,
-  destinationTokenAddress: string,
-  simulatedRelayerAddress: string
+  deposit: Deposit,
+  amountToFill: BN,
+  relayerAddress: string,
+  _relayerBalanceForToken?: BN
 ): Promise<PopulatedTransaction> {
-  // Populate and return an unsigned tx as per the given spoke pool
-  // NOTE: 0xBb23Cd0210F878Ea4CcA50e9dC307fb0Ed65Cf6B is a dummy address
-  return await spokePool.populateTransaction.fillRelay(
-    "0xBb23Cd0210F878Ea4CcA50e9dC307fb0Ed65Cf6B",
-    "0xBb23Cd0210F878Ea4CcA50e9dC307fb0Ed65Cf6B",
-    destinationTokenAddress,
-    "10",
-    "10",
-    "1",
-    "1",
-    "1",
-    "1",
-    "1",
-    [],
-    MAX_BIG_INT,
-    { from: simulatedRelayerAddress }
-  );
+  // We should check that the relayer has enough balance to facilitate this
+  // transaction before we populate it.
+  const relayerBalanceForToken =
+    _relayerBalanceForToken ?? (await getTokenBalance(relayerAddress, deposit.originToken, spokePool.provider));
```

Comment:
I just had a thought: do we really want to perform the check on `depositAmount` versus `_relayerBalanceForToken` in the sdk? Why can't we do this check in the relayer or the API, and then based on the result of that comparison, decide whether to call into the sdk?

Reply:
From the relayer perspective I am OK if it's not done here, because the relayer already tracks its balance before calling in to this method. If it remains here then I'll just supply `fillAmount` to ensure that the RPC call is never triggered. The RPC balance check is actually not useful for the relayer in any case, since it's balance can change before the fill is executed (i.e. if this is filled in a multicall and other fills come before it).

Reply:
We can drop it. Ultimately the relayer will not incur any additional RPC calls since its passing the balance in - so this is more of a synchronous validation for it.

Reply:
https://github.com/across-protocol/sdk-v2/pull/427/commits/f62328a66fcbd8c63b7faf1f760e5aa5fec81619